data-prepper-plugins/aggregate-processor/src/main/java/com/amazon/dataprepper/plugins/processor/aggregate/AggregateGroupManager.java

```diff
@@ -15,22 +16,14 @@
 class AggregateGroupManager {
 
     private final Map<AggregateIdentificationKeysHasher.IdentificationHash, AggregateGroup> allGroups = Maps.newConcurrentMap();
-    private final int windowDuration;
+    private final Duration windowDuration;
 
     AggregateGroupManager(final int windowDuration) {
-        this.windowDuration = windowDuration;
+        this.windowDuration = Duration.ofSeconds(windowDuration);
     }
 
     AggregateGroup getAggregateGroup(final AggregateIdentificationKeysHasher.IdentificationHash identificationHash) {
-        final AggregateGroup aggregateGroup = allGroups.get(identificationHash);
-
-        if (aggregateGroup != null) {
-            return aggregateGroup;
-        }
-
-        final AggregateGroup newAggregateGroup = new AggregateGroup();
-        allGroups.put(identificationHash, newAggregateGroup);
-        return newAggregateGroup;
+        return allGroups.computeIfAbsent(identificationHash, (hash) -> new AggregateGroup());
```

Comment:
I like the simplification.

Reply:
It is also atomic per the `ConcurrentMap` Javadocs.